src/wp-includes/default-filters.php

```diff
@@ -762,7 +763,9 @@
 // Update ignoredHookedBlocks postmeta for wp_navigation post type.
 add_filter( 'rest_pre_insert_wp_navigation', 'update_ignored_hooked_blocks_postmeta' );
 
-// Inject hooked blocks into the wp_navigation post type REST response.
+// Inject hooked blocks into the Posts endpoint REST response for some given post types.
+add_filter( 'rest_prepare_page', 'insert_hooked_blocks_into_rest_response', 10, 2 );
+add_filter( 'rest_prepare_post', 'insert_hooked_blocks_into_rest_response', 10, 2 );
```

Comment:
This follows the precedent of `rest_prepare_wp_navigation` below. However, it has the unfortunate side effect that Block Hooks aren't automatically applied to user-defined CPTs; those would need to be added manually (via `add_filter( 'rest_prepare_my_cpt', ... )`). (AFAIK, there's no "generic" `rest_prepare` filter that could cover all post types.)

I wonder if we should run `insert_hooked_blocks_into_rest_response` from inside the Posts controller instead, in order to insert hooked blocks into every CPT that has `show_in_rest` set to `true` 🤔 cc/ @gziolo

Reply:
It would be worthwhile to explore supporting other types than `post`, `page`, and `wp_navigation`.

Reply:
I've filed https://core.trac.wordpress.org/ticket/62715 to keep track of this.